### PR TITLE
Restore bindToContext functionality

### PR DESCRIPTION
**Pull request by markfields**
After discussing the challenges being faced by our internal partners integrating the 2.0.0-internal.1.1.x release due to the removal of `bindToContext` (see #10627), we're adding it back in.  They are tracking the work required to fully migrate off the request pattern to using handles, which will render this fully obsolete and at that point we will remove it.

## Does this introduce a breaking change?

No... it undoes a breaking change, such that this change is additive.

Same change going into release/v2int/1.1 is #12048

### CODE REVIEW

**Review by agarwal-navin — APPROVED** [review]

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 1]
```diff
@@ -2683,7 +2685,7 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             // Please note that this does not change file format, so it can be disabled in the future if this
             // optimization no longer makes sense (for example, batch compression may make it less appealing).
             if (this._flushMode === FlushMode.TurnBased && type === ContainerMessageType.Attach &&
-                this.mc.config.getBoolean("Fluid.ContainerRuntime.disableAttachOpReorder") !== true) {
+                this.mc.config.getBoolean("Fluid.ContainerRuntime.enableAttachOpReorder") === true) {
```
Comment (agarwal-navin) [inline packages/runtime/container-runtime/src/containerRuntime.ts:2688]: nit: should we use the same config name so as to not change it later:
`this.mc.config.getBoolean("Fluid.ContainerRuntime.disableAttachOpReorder") === false`

Comment (agarwal-navin) [reply packages/runtime/container-runtime/src/containerRuntime.ts:2688]: Also, this should probably be done is a separate PR to keep its context separate.

Comment (markfields) [reply packages/runtime/container-runtime/src/containerRuntime.ts:2688]: I started that way but figured there's no hard dependency anywhere on that setting name, and Vlad is going to revisit this code anyway. So I opted for the more natural phrasing.

And sure, I can do a separate PR.

**Comment by markfields** [comment]
Update: Planning to hold on this until next week when FF/Bohemia stakeholders have a chance to discuss and agree upon a long term plan for this API (i.e. when we can remove it for good)

**Pull request merged**